apps/advisor/pkg/app/checktyperegisterer/checktyperegisterer.go

```diff
@@ -73,18 +81,25 @@ func (r *Runner) Run(ctx context.Context) error {
 			},
 		}
 		id := obj.GetStaticMetadata().Identifier()
-		_, err := r.client.Create(ctx, id, obj, resource.CreateOptions{})
-		if err != nil {
-			if errors.IsAlreadyExists(err) {
-				// Already exists, update
-				_, err = r.client.Update(ctx, id, obj, resource.UpdateOptions{})
-				if err != nil {
-					return err
-				} else {
-					continue
+		for i := 0; i < r.retryAttempts; i++ {
+			_, err := r.client.Create(ctx, id, obj, resource.CreateOptions{})
+			if err != nil {
+				if errors.IsAlreadyExists(err) {
+					// Already exists, update
+					_, err = r.client.Update(ctx, id, obj, resource.UpdateOptions{})
+					if err != nil {
+						// Ignore the error, it's probably due to a race condition
+						r.log.Error("Error updating check type", "error", err)
+					} else {
+						continue
+					}
+				}
+				r.log.Error("Error creating check type, retrying", "error", err, "attempt", i+1)
```

Comment:
Couldn't these end up logging the same error? Should the error from update maybe be `updateErr` or `innerErr` to differentiate?

Reply:
Right, we also don't need to retry if it fails when updating and I'm also missing the early return if the create doesn't fail.